plugins/coverbrowser.koplugin/main.lua

```diff
@@ -161,58 +163,143 @@ function CoverBrowser:addToMainMenu(menu_items)
     -- add Mosaic / Detailed list mode settings to File browser Settings submenu
     -- next to Classic mode settings
     if menu_items.filebrowser_settings == nil then return end
-    table.insert (menu_items.filebrowser_settings.sub_item_table, 4, {
+    table.insert (menu_items.filebrowser_settings.sub_item_table, 5, {
         text = _("Mosaic and detailed list settings"),
         separator = true,
         sub_item_table = {
             {
-                text = _("Items per page"),
-                help_text = _([[This sets the number of files and folders per page in display modes other than classic.]]),
+                text_func = function()
+                    local nb_cols_portrait = tonumber(BookInfoManager:getSetting("nb_cols_portrait") or 3)
+                    local nb_rows_portrait = tonumber(BookInfoManager:getSetting("nb_rows_portrait") or 3)
+                    return T(_("Items per page in mosaic portrait mode: %1 × %2"), nb_cols_portrait, nb_rows_portrait)
+                end,
                 -- Best to not "keep_menu_open = true", to see how this apply on the full view
-                callback = function()
+                callback = function(touchmenu_instance)
+                    local nb_cols_portrait = tonumber(BookInfoManager:getSetting("nb_cols_portrait") or 3)
+                    local nb_rows_portrait = tonumber(BookInfoManager:getSetting("nb_rows_portrait") or 3)
+                    local DoubleSpinWidget = require("/ui/widget/doublespinwidget")
+                    local widget = DoubleSpinWidget:new{
+                        title_text = _("Mosaic portrait mode"),
+                        width_factor = 0.6,
+                        left_text = _("Columns"),
+                        left_value = nb_cols_portrait,
+                        left_min = 2,
+                        left_max = 5,
+                        left_default = 3,
+                        left_precision = "%01d",
+                        right_text = _("Rows"),
+                        right_value = nb_rows_portrait,
+                        right_min = 2,
+                        right_max = 5,
+                        right_default = 3,
+                        right_precision = "%01d",
+                        keep_shown_on_apply = true,
+                        callback = function(left_value, right_value)
+                            BookInfoManager:saveSetting("nb_cols_portrait", left_value)
+                            BookInfoManager:saveSetting("nb_rows_portrait", right_value)
+                            self.layout_updated = left_value ~= nb_cols_portrait or right_value ~= nb_rows_portrait
+                            file_chooser.refresh_covers = false
```

Comment:
Maybe name it `no_refresh_covers`, only existing and `true` in this transition phase, otherwise nil.

Reply:
Already done in my testing, it fixes rotation without any changes to the rotation code.